### PR TITLE
ROX-16495: Installation instructions for PostgreSQL tech preview

### PR DESCRIPTION
**Pull request by connorgorman**
Version(s):

Merge to rhacs-docs; cherry pick to rhacs-docs-4.0

Issue:

[ROX-16495](https://issues.redhat.com/browse/ROX-16495)

Link to docs preview:

Is this a good spot for this page?
https://58668--docspreview.netlify.app/openshift-acs/latest/installing/installing_ocp/install-central-ocp.html#install-central-operator-tech-preview_install-central-ocp

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/install-central-operator-tech-preview.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+. Using the terminal, create a password secret in the newly created project. password.txt is the password for the superuser of the provisioned database.
+[source,terminal]
+----
+$ oc create secret generic external-db-password --from-file=password=password.txt
+----
+. Under the *Provided APIs* section, select *Central*. Click *Create Central*.
+. Enter a name for your `Central` custom resource and add any labels you want to apply.
+. Navigate to *Central Component Settings* > *Central DB Settings*.
+. Underneath *Administrator Password* specify the referenced secret as `external-db-password` (or the secret name of the password created above).
+. Specify the connection string in keyword/value format as described https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING[here]. e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`
```
Comment (theencee) [inline modules/install-central-operator-tech-preview.adoc:50]:
```suggestion
. Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING[here]. e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`
```

File: modules/install-central-operator-tech-preview.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+. Underneath *Persistence* > *PersistentVolumeClaim* > *Claim Name* remove `central-db`.
+. If necessary, a Certificate Authority can be specified in order for Central to trust the database certificate. This can be added by navigating to the YAML view and adding a TLS block underneath the top-level spec:
+[source,yaml]
+----
```
Comment (theencee) [inline modules/install-central-operator-tech-preview.adoc:54]:
```suggestion
+
[source,yaml]
----
```

File: modules/install-central-operator-tech-preview.adoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+.Prerequisites
+* You must be using {ocp} 4.6 or later.
+* Provision a database that supports PostgreSQL 13 that will be used solely for ACS (shared infrastructure is currently unsupported).
+
+[NOTE]
+====
+* A multi-tenant database is currently unsupported.
+* Superuser role is required.
+* PostgreSQL 14 or 15 _may_ work, but is not officially supported.
```
[the remark this answers is not in the document]
Comment (connorgorman) [reply modules/install-central-operator-tech-preview.adoc:27]: will do

File: modules/install-central-operator-tech-preview.adoc [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+* PgBouncer is currently unsupported.
+====
+
+.Procedure
+. On the {ocp} web console, navigate to the *Operators* -> *Installed Operators* page.
+. Select the {product-title} Operator from the list of installed Operators.
+. If you have installed the Operator in the recommended namespace, {ocp} lists the project as `rhacs-operator`. Select *Project: rhacs-operator* -> *Create project*.
+[WARNING]
```
[the remark this answers is not in the document]
Comment (connorgorman) [reply modules/install-central-operator-tech-preview.adoc:35]: will do

**Comment by connorgorman** [comment]
CC: @gaurav-nelson

File: modules/install-central-operator-tech-preview.adoc [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+The main component of {product-title} is called Central. You can install Central on {ocp} by using the `Central` custom resource. You deploy Central only once, and you can monitor multiple separate clusters by using the same Central installation.
+
+[IMPORTANT]
+====
+When you install {product-title} for the first time, you must first install the `Central` custom resource because the `SecuredCluster` custom resource installation is dependent on certificates that Central generates.
+====
+
+.Prerequisites
+* You must be using {ocp} 4.6 or later.
+* Provision a database that supports PostgreSQL 13 that will be used solely for ACS.
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:21]:
```suggestion
* Provision a database that supports PostgreSQL 13 that will be used solely for {product-title-short}.
```

File: modules/install-central-operator-tech-preview.adoc [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
++
+[source,terminal]
+----
+$ oc create secret generic external-db-password --from-file=password=password.txt
+----
+. Return to the {product-title} operator page in the {ocp} web console. Under the *Provided APIs* section, select *Central*. Click *Create Central*.
+. Enter a name for your `Central` custom resource and add any labels you want to apply.
+. Navigate to *Central Component Settings* > *Central DB Settings*.
+. Underneath *Administrator Password* specify the referenced secret as `external-db-password` (or the secret name of the password created above).
+. Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING[here]. e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:53]: link issue, see my comments on https://github.com/openshift/openshift-docs/pull/58655

File: installing/installing_ocp/install-central-ocp.adoc [hunk 7]
```diff
@@ -24,11 +24,13 @@ You can install Central on your {ocp} or Kubernetes cluster by using one of the
 include::modules/install-acs-operator.adoc[leveloffset=+2]
 
 include::modules/install-central-operator.adoc[leveloffset=+2]
+include::modules/install-central-operator-tech-preview.adoc[leveloffset=+2]
```
Comment (kcarmichael08) [inline installing/installing_ocp/install-central-ocp.adoc:27]: This looks like a good place to put it.

File: modules/install-central-operator-tech-preview.adoc [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+[role="_abstract"]
+The main component of {product-title} is called Central. You can install Central on {ocp} by using the `Central` custom resource. You deploy Central only once, and you can monitor multiple separate clusters by using the same Central installation.
+
+[IMPORTANT]
+====
+When you install {product-title} for the first time, you must first install the `Central` custom resource because the `SecuredCluster` custom resource installation is dependent on certificates that Central generates.
+====
+
+.Prerequisites
+* You must be using {ocp} 4.6 or later.
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:20]: This just got changed recently to 4.10 or later so you might get a merge conflict. As of ACS 4.0, it's 4.10

File: modules/install-central-operator-tech-preview.adoc [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
++
+[WARNING]
+====
+* If you have installed the Operator in a different namespace, {ocp} shows the name of that namespace rather than `rhacs-operator`.
+* You must install the {product-title} `Central` custom resource in its own project and not in the `rhacs-operator` and `openshift-operator` projects, or in the project in which you have installed the {product-title} Operator.
+====
+. Enter the new project name (for example, `stackrox`), and click *Create*. Red Hat recommends that you use `stackrox` as the project name.
+. Create a password secret by navigating to the *Workloads* -> *Secrets* page on the {ocp} web console. Create a `Key/Value secret` with the key `password` and the value being a plaintext file containing the password for the superuser of the provisioned database.
+Alternatively, using the terminal, create a password secret in the newly created project. password.txt is the password for the superuser of the provisioned database.
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:43]:
```suggestion
Alternatively, using the terminal, create a password secret in the newly created project. password.txt is the password for the superuser of the provisioned database. Enter the following command:
```

File: modules/install-central-operator-tech-preview.adoc [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
++
+[source,terminal]
+----
+$ oc create secret generic external-db-password --from-file=password=password.txt
+----
+. Return to the {product-title} operator page in the {ocp} web console. Under the *Provided APIs* section, select *Central*. Click *Create Central*.
+. Enter a name for your `Central` custom resource and add any labels you want to apply.
+. Navigate to *Central Component Settings* > *Central DB Settings*.
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:51]:
```suggestion
. Navigate to *Central Component Settings* -> *Central DB Settings*.
```

Comment (kcarmichael08) [reply modules/install-central-operator-tech-preview.adoc:51]: (when transformed to HTML this creates an arrow)

File: modules/install-central-operator-tech-preview.adoc [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+. Underneath *Administrator Password* specify the referenced secret as `external-db-password` (or the secret name of the password created above).
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:52]:
```suggestion
. For *Administrator Password*, specify the referenced secret as `external-db-password` (or the secret name of the password created previously).
```

File: modules/install-central-operator-tech-preview.adoc [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+. Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described in the PostgreSQL documentation in "Additional resources". e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:53]:
```suggestion
. For *Connection String (Technology Preview)*, specify the connection string in keyword/value format as described in the PostgreSQL documentation in "Additional resources".; for example, `host=<host> port=5432 user=postgres sslmode=verify-ca`
```

File: modules/install-central-operator-tech-preview.adoc [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+. Underneath *Persistence* > *PersistentVolumeClaim* > *Claim Name* remove `central-db`.
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:54]:
```suggestion
. For *Persistence* -> *PersistentVolumeClaim* -> *Claim Name*, remove `central-db`.
```

File: modules/install-central-operator-tech-preview.adoc [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+. If you have installed the Operator in the recommended namespace, {ocp} lists the project as `rhacs-operator`. Select *Project: rhacs-operator* -> *Create project*.
+
++
+[WARNING]
+====
+* If you have installed the Operator in a different namespace, {ocp} shows the name of that namespace rather than `rhacs-operator`.
+* You must install the {product-title} `Central` custom resource in its own project and not in the `rhacs-operator` and `openshift-operator` projects, or in the project in which you have installed the {product-title} Operator.
+====
+. Enter the new project name (for example, `stackrox`), and click *Create*. Red Hat recommends that you use `stackrox` as the project name.
+. Create a password secret by navigating to the *Workloads* -> *Secrets* page on the {ocp} web console. Create a `Key/Value secret` with the key `password` and the value being a plaintext file containing the password for the superuser of the provisioned database.
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:42]:
```suggestion
. Create a password secret by navigating to the *Workloads* -> *Secrets* page on the {ocp} web console. Create a `Key/Value secret` with the key `password` and the value of a plaintext file containing the password for the superuser of the provisioned database.
```

File: modules/install-central-operator-tech-preview.adoc [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+----
+$ oc create secret generic external-db-password --from-file=password=password.txt
+----
+. Return to the {product-title} operator page in the {ocp} web console. Under the *Provided APIs* section, select *Central*. Click *Create Central*.
+. Enter a name for your `Central` custom resource and add any labels you want to apply.
+. Navigate to *Central Component Settings* > *Central DB Settings*.
+. Underneath *Administrator Password* specify the referenced secret as `external-db-password` (or the secret name of the password created above).
+. Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described in the PostgreSQL documentation in "Additional resources". e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`
+. Underneath *Persistence* > *PersistentVolumeClaim* > *Claim Name* remove `central-db`.
+. If necessary, a Certificate Authority can be specified in order for Central to trust the database certificate. This can be added by navigating to the YAML view and adding a TLS block underneath the top-level spec:
```
Comment (kcarmichael08) [inline modules/install-central-operator-tech-preview.adoc:55]:
```suggestion
. If necessary, you can specify a Certificate Authority for Central to trust the database certificate. To add this, navigate to the YAML view and add a TLS block under the top-level spec, as shown in the following example:
```

**Review by kcarmichael08 — COMMENTED** [review]
This looks great! I had some minor suggestions based on the Style Guides that we have to follow. TY so much for submitting this and making it easy for us to include!

**Comment by kcarmichael08** [comment]
PS - Before we can request to merge, all commits have to be squashed into one (per OpenShift doc repo guidelines) so if there aren't any more changes, please squash.

File: modules/install-central-operator-tech-preview.adoc [hunk 16]
```diff
@@ -0,0 +1,78 @@
+// Module included in the following assemblies:
+//
+// * installing/install-ocp-operator.adoc
+:_module-type: PROCEDURE
+[id="install-central-operator-tech-preview_{context}"]
+= Installing Central with an external database using the Operator method
```
Comment (gaurav-nelson) [inline modules/install-central-operator-tech-preview.adoc:6]: Installing Central with an external database using the {product-title-short} Operator

File: modules/install-central-operator-tech-preview.adoc [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+The main component of {product-title} is called Central. You can install Central on {ocp} by using the `Central` custom resource. You deploy Central only once, and you can monitor multiple separate clusters by using the same Central installation.
+
+[IMPORTANT]
+====
+When you install {product-title} for the first time, you must first install the `Central` custom resource because the `SecuredCluster` custom resource installation is dependent on certificates that Central generates.
+====
+
+.Prerequisites
+* You must be using {ocp} 4.10 or later.
+* Provision a database that supports PostgreSQL 13 that will be used solely for {product-title-short}.
```
Comment (gaurav-nelson) [inline modules/install-central-operator-tech-preview.adoc:21]: Create a PostgreSQL 13 database dedicated exclusively to {product-title-short}.

File: modules/install-central-operator-tech-preview.adoc [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+
+[NOTE]
+====
+* A multi-tenant database is currently unsupported.
+* Superuser role is required with permissions to create and delete databases.
+* PgBouncer is currently unsupported.
+====
+
```
Comment (gaurav-nelson) [inline modules/install-central-operator-tech-preview.adoc:29]:
```suggestion
[NOTE]
====
{product-title-short} 4.0 does not support a multi-tenant database and PgBouncer.
====

.Prerequisites
* You must have a superuser role with permissions to create and delete databases.

```

**Pull request merged**